src/api/app/controllers/webui/repositories_controller.rb

```diff
@@ -241,12 +244,42 @@ def remove_flag
       @main_object.store
       format.html { redirect_to(action: :index, project: params[:project], package: params[:package]) }
       format.js do
-        switch_to_webui2
         render 'change_flag'
       end
     end
   end
 
+  # POST flag/manipulate/:project(/:package)
+  # TODO when removing bento, remove the extra manipulate from the route, for
+  # now we need to avoid the clash with create_flag
+  def manipulate_flag
+    unless request.xhr?
+      redirect_to(action: :index, project: params[:project], package: params[:package])
+      return
+    end
+    required_parameters :flag, :command
+    set_webui2_views
+    authorize @main_object, :update?
+    architecture  = Architecture.from_cache!(params[:architecture]) unless params[:architecture].nil?
+    flag_type = params[:flag]
+    if params[:command] == 'remove'
+      @main_object.flags.of_type(flag_type).where(repo: params[:repository], architecture: architecture).delete_all
+    elsif params[:command] == 'set-disable'
+      @main_object.flags.create(flag: flag_type, repo: params[:repository], architecture: architecture, status: 'disable')
+    elsif params[:command] == 'set-enable'
+      @main_object.flags.create(flag: flag_type, repo: params[:repository], architecture: architecture, status: 'enable')
+    end
+    # FIXME: This should happen in Flag or even better in Project
+    @main_object.store
```

Comment:
This store is rather expensive for larger projects, but fixing the performance of _meta rendering benefits us in more places